File: web/src/api/campusProduct.js
import service from '@/utils/request'

export const getCampusProductList = (params) => {
  return service({
    url: '/campusProduct/getCampusProductList',
    method: 'get',
    params
  })
}

export const findCampusProduct = (params) => {
  return service({
    url: '/campusProduct/findCampusProduct',
    method: 'get',
    params
  })
}

export const updateCampusProductStatus = (data) => {
  return service({
    url: '/campusProduct/updateCampusProductStatus',
    method: 'post',
    data
  })
}

File: server/router/campus/campus_product.go
package campus

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CampusProductRouter struct{}

func (r *CampusProductRouter) InitCampusProductRouter(Router *gin.RouterGroup) {
	productRouter := Router.Group("campusProduct").Use(middleware.OperationRecord())
	productRouterWithoutRecord := Router.Group("campusProduct")
	{
		productRouter.POST("updateCampusProductStatus", campusProductApi.UpdateCampusProductStatus)
	}
	{
		productRouterWithoutRecord.GET("getCampusProductList", campusProductApi.GetCampusProductList)
		productRouterWithoutRecord.GET("findCampusProduct", campusProductApi.FindCampusProduct)
	}
}
